### ISSUE TITLE
Selenium 3.3.1 & Firefox 52.0 Unexpectedly aborting tests 

### ISSUE THREAD

**Issue by monsonjeremy** [mozilla/geckodriver#552]
## Firefox Version
52.0 32 bit

## Platform
Windows and OSX

## Description of issue
When using Selenium 3.3.1 & FF 52.0 I am unable to run any kind of tests. 
I found this issue when trying to find elements on the page or scroll down the page in my own tests. My temporary workaround has been to downgrade to selenium 2.53.6 and Firefox 45.6.

## Steps to reproduce -

geckodriver.log
```
1489700295614	geckodriver	INFO	Listening on 127.0.0.1:54810
1489700296152	mozprofile::profile	INFO	Using profile path C:\Users\USERNAME\AppData\Local\Temp\rust_mozprofile.8PPdlKjiQW1X
1489700296163	geckodriver::marionette	INFO	Starting browser C:\Program Files (x86)\Mozilla Firefox\firefox.exe
1489700296172	geckodriver::marionette	INFO	Connecting to Marionette on localhost:54814
1489700296971	Marionette	INFO	Listening on port 54814
1489700297308	Marionette	WARN	TLS certificate errors will be ignored for this session
1489700300662	Marionette	INFO	New connections will no longer be accepted
[Child 8172] ###!!! ABORT: Aborting on channel error.: file c:/builds/moz2_slave/m-rel-w32-00000000000000000000/build/src/ipc/glue/MessageChannel.cpp, line 2143
```

#Problem.py

```
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
 
driver = webdriver.Firefox()
driver.maximize_window()
 
driver.get("http://www.google.com")
 
search_field = driver.find_element_by_id("lst-ib")
search_field.clear()
 
search_field.send_keys("Selenium WebDriver Interview questions")
search_field.submit()
 
lists= driver.find_elements_by_class_name("_Rm")

print ("Found " + str(len(lists)) + "searches:")
 
i=0
for listitem in lists:
   print (listitem)
   i=i+1
   if(i>10):
      break
 
driver.quit()

```

**Comment by wgracie** [comment]
I also encountered this problem.
Env:
Selenium: selenium-server-standalone-3.3.1.jar
Firefox: firefox-52.0.tar.bz2
geckodriver: geckodriver-v0.15.0-linux64.tar.gz

Logs:
1489732530896	Marionette	INFO	New connections will no longer be accepted
###!!! [Child][MessageChannel] Error: (msgtype=0x3E0003,name=PCompositable::Msg_Destroy) Channel error: cannot send/recv
[Child 177] ###!!! ABORT: Aborting on channel error.: file /builds/slave/m-rel-l64-00000000000000000000/build/src/ipc/glue/MessageChannel.cpp, line 2143
###!!! [Child][MessageChannel] Error: (msgtype=0xE40003,name=PTexture::Msg_Destroy) Channel error: cannot send/recv
###!!! [Child][MessageChannel] Error: (msgtype=0xE40003,name=PTexture::Msg_Destroy) Channel error: cannot send/recv
[Child 177] ###!!! ABORT: Aborting on channel error.: file /builds/slave/m-rel-l64-00000000000000000000/build/src/ipc/glue/MessageChannel.cpp, line 2143
###!!! [Child][MessageChannel] Error: (msgtype=0x3E0003,name=PCompositable::Msg_Destroy) Channel error: cannot send/recv
06:35:31.026 INFO - Done: [delete session: 25fa2682-d2d2-4c36-98b8-6b76fed12d38]

**Comment by nex0ma** [comment]
yes, not worked, but started, that is plus

    //    click on partialLinkText("Аналитические панели")
    //    remote.findElements(By.partialLinkText(""""Аналитические панели"""")).get(0).click()
    //    remote.findElements(By.partialLinkText(""""Аналитические панели"""")).get(0).click()
    //    remote.findElements(By.xpath("""//*[@id="main-navbar"]/ul[3]/li[1]/a""")).get(0).click()
    //    remote.findElements(By.xpath("""#main-navbar > ul.nav.nav-tabs.navbar-left.hidden-xs.hidden-sm.nobottomborder > li.active > a""")).get(0).click()
    //    remote.findElements(By.xpath("""//a[@href="/Dashboards/CompetitiveDirections"]""")).get(0).click()

so, chrome, phantomjs, iexplorer11 (all in remotewebdriver mode) worked

libraryDependencies += "org.seleniumhq.selenium" % "selenium-java" % "3.3.1"
libraryDependencies += "org.seleniumhq.selenium" % "selenium-firefox-driver" % "3.3.1"

**Comment by nex0ma** [comment]
for click

[info]   org.openqa.selenium.ElementNotVisibleException: Build info: version: '3.3.1', revision: '5234b325d5', time: '2017-03-10 09:10:29 +0000'
[info] System info: host: '***', ip: '***', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.8.0_11'
[info] Driver info: org.openqa.selenium.firefox.FirefoxDriver
[info] Capabilities [{moz:profile=C:\Users\R***\AppData\Local\Temp\rust_mozprofile.Ib8NSHOm7UU2, rotatable=false, timeouts={implicit=0, page load=300000, script=30000}, pageLoadStrategy=normal, platform=ANY, specificationLevel=0, moz:accessibilityChecks=false, acceptInsecureCerts=false, browserVersion=52.0, platformVersion=6.1, moz:processID=22396, browserName=firefox, platformName=windows_nt}]
[info] Session ID: b8cd38e9-f1af-401e-83e2-6446bf076eba
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[info]   at java.lang.reflect.Constructor.newInstance(Constructor.java:408)
[info]   at org.openqa.selenium.remote.http.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:133)
[info]   at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:99)
[info]   at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:43)
[info]   at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:163)
[info]   at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:82)
[info]   at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:604)

**Comment by nex0ma** [comment]
but on Linux Opensuse Thumbleweed 64 worked (with remotewebdriver connection)!

**Comment by Candide7** [comment]
Same problem with Win10, Selenium 3.3.1, Gecko 0.15 and firefox 52

`1489917396641	geckodriver	INFO	Listening on 127.0.0.1:21138
1489917397153	mozprofile::profile	INFO	Using profile path X:\TEMP\rust_mozprofile.iuBb0nK9qKKi
1489917397161	geckodriver::marionette	INFO	Starting browser C:\Program Files (x86)\Mozilla Firefox\firefox.exe with args []
1489917397171	geckodriver::marionette	INFO	Connecting to Marionette on localhost:62914
1489917398534	Marionette	INFO	Listening on port 62914
mars 19, 2017 10:56:41 AM org.openqa.selenium.remote.ProtocolHandshake createSession
INFOS: Detected dialect: W3C
1489917402992	Marionette	INFO	New connections will no longer be accepted
[Child 1664] ###!!! ABORT: Aborting on channel error.: file c:/builds/moz2_slave/m-rel-w32-00000000000000000000/build/src/ipc/glue/MessageChannel.cpp, line 2143
mars 19, 2017 10:56:44 AM org.openqa.selenium.os.UnixProcess destroy
GRAVE: Unable to kill process with PID 9476
`

**Comment by whimboo** [comment]
When you all observe the tests, can you notice a hang of the currently selected tab? The above shows a IPC error between the chrome and content process, which will end-up in a shutdown of Firefox.

**Comment by Candide7** [comment]
"can you notice a hang" --> Can you precise your question please?

**Comment by whimboo** [comment]
Hm, when I check the log from @Candide7 again I can see by the timestamps that Firefox was running only ~5s. It smells like a startup crash. Can you please try with a Nightly build of Firefox again?

**Comment by Candide7** [comment]
Firefox doesn't crash by its own. But yes I can test again with an update.
This version may'be? https://www.mozilla.org/en-US/firefox/53.0beta/releasenotes/

I'm really surprised this bug can't be reproduced since several people face the same issue which seems very version dependent.

**Comment by meteboyaci** [comment]
Hi everyone,
I have been dealing with the same issue for several days. The error message that i faced with is mentioned below. I tried different versions of firefox (firefox 47, 49, 51, 52) but i could not get rid of this issue.
My current settings are: Selenium-java 3.3.1 & Firefox 52.0 & geckodriver-v0.15.0-win64
org.openqa.selenium.firefox.NotConnectedException: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
URL":null,"defaultLocale":{"name":"Default","description":"The default theme.","creator":"Mozilla","homepageURL":null,"contributors":["Mozilla Contributors"]},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"C:\\Program Files\\Mozilla Firefox\\browser\\extensions\\{972ce4c6-7e08-4474-a285-3208198ce6fd}.xpi","installDate":1483666073000,"updateDate":1483666073000,"applyBackgroundUpdates":1,"skinnable":true,"size":19702,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":true,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"51.0","maxVersion":"51.0"}],"targetPlatforms":[],"seen":true,"dependencies":[],"hasEmbeddedWebExtension":false}
1490267391957	addons.xpi	DEBUG	getModTime: Recursive scan of {972ce4c6-7e08-4474-a285-3208198ce6fd}
1490267391958	DeferredSave.extensions.json	DEBUG	Save changes
1490267391958	addons.xpi	DEBUG	Updating database with changes to installed add-ons
1490267391958	addons.xpi-utils	DEBUG	Updating add-on states
1490267391960	addons.xpi-utils	DEBUG	Writing add-ons list
1490267391966	addons.xpi	DEBUG	Registering manifest for C:\Program Files\Mozilla Firefox\browser\features\aushelper@mozilla.org.xpi
1490267391967	addons.xpi	DEBUG	Calling bootstrap method startup on aushelper@mozilla.org version 1.0
1490267391968	addons.xpi	DEBUG	Registering manifest for C:\Program Files\Mozilla Firefox\browser\features\e10srollout@mozilla.org.xpi
1490267391969	addons.xpi	DEBUG	Calling bootstrap method startup on e10srollout@mozilla.org version 1.6
1490267391969	addons.xpi	DEBUG	Registering manifest for C:\Program Files\Mozilla Firefox\browser\features\firefox@getpocket.com.xpi
1490267391970	addons.xpi	DEBUG	Calling bootstrap method startup on firefox@getpocket.com version 1.0.5
1490267391970	addons.xpi	DEBUG	Registering manifest for C:\Program Files\Mozilla Firefox\browser\features\webcompat@mozilla.org.xpi
1490267391971	addons.xpi	DEBUG	Calling bootstrap method startup on webcompat@mozilla.org version 1.0
1490267391972	addons.manager	DEBUG	Registering shutdown blocker for XPIProvider
1490267391972	addons.manager	DEBUG	Provider finished startup: XPIProvider
1490267391973	addons.manager	DEBUG	Starting provider: LightweightThemeManager
1490267391973	addons.manager	DEBUG	Registering shutdown blocker for LightweightThemeManager
1490267391973	addons.manager	DEBUG	Provider finished startup: LightweightThemeManager
1490267391974	addons.manager	DEBUG	Starting provider: GMPProvider
1490267391984	addons.manager	DEBUG	Registering shutdown blocker for GMPProvider
1490267391984	addons.manager	DEBUG	Provider finished startup: GMPProvider
1490267391984	addons.manager	DEBUG	Starting provider: PluginProvider
1490267391984	addons.manager	DEBUG	Registering shutdown blocker for PluginProvider
1490267391984	addons.manager	DEBUG	Provider finished startup: PluginProvider
1490267391985	addons.manager	DEBUG	Completed startup sequence
1490267393150	addons.manager	DEBUG	Starting provider: <unnamed-provider>
1490267393150	addons.manager	DEBUG	Registering shutdown blocker for <unnamed-provider>
1490267393151	addons.manager	DEBUG	Provider finished startup: <unnamed-provider>
1490267393152	DeferredSave.extensions.json	DEBUG	Starting write
1490267393628	addons.repository	DEBUG	No addons.json found.
1490267393628	DeferredSave.addons.json	DEBUG	Save changes
1490267393632	DeferredSave.addons.json	DEBUG	Starting timer
1490267393682	addons.manager	DEBUG	Starting provider: PreviousExperimentProvider
1490267393682	addons.manager	DEBUG	Registering shutdown blocker for PreviousExperimentProvider
1490267393682	addons.manager	DEBUG	Provider finished startup: PreviousExperimentProvider
1490267393684	DeferredSave.extensions.json	DEBUG	Write succeeded
1490267393684	addons.xpi-utils	DEBUG	XPI Database saved, setting schema version preference to 19
1490267393684	DeferredSave.addons.json	DEBUG	Starting write
1490267393714	DeferredSave.addons.json	DEBUG	Write succeeded
org.openqa.selenium.firefox.NotConnectedException: Unable to connect to host 127.0.0.1 on port 7055 after 45000 ms. Firefox console output:
URL":null,"defaultLocale":{"name":"Default","description":"The default theme.","creator":"Mozilla","homepageURL":null,"contributors":["Mozilla Contributors"]},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"C:\\Program Files\\Mozilla Firefox\\browser\\extensions\\{972ce4c6-7e08-4474-a285-3208198ce6fd}.xpi","installDate":1483666073000,"updateDate":1483666073000,"applyBackgroundUpdates":1,"skinnable":true,"size":19702,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":true,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"51.0","maxVersion":"51.0"}],"targetPlatforms":[],"seen":true,"dependencies":[],"hasEmbeddedWebExtension":false}
1490267391957	addons.xpi	DEBUG	getModTime: Recursive scan of {972ce4c6-7e08-4474-a285-3208198ce6fd}
1490267391958	DeferredSave.extensions.json	DEBUG	Save changes
1490267391958	addons.xpi	DEBUG	Updating database with changes to installed add-ons
1490267391958	addons.xpi-utils	DEBUG	Updating add-on states
1490267391960	addons.xpi-utils	DEBUG	Writing add-ons list
1490267391966	addons.xpi	DEBUG	Registering manifest for C:\Program Files\Mozilla Firefox\browser\features\aushelper@mozilla.org.xpi
1490267391967	addons.xpi	DEBUG	Calling bootstrap method startup on aushelper@mozilla.org version 1.0
1490267391968	addons.xpi	DEBUG	Registering manifest for C:\Program Files\Mozilla Firefox\browser\features\e10srollout@mozilla.org.xpi
1490267391969	addons.xpi	DEBUG	Calling bootstrap method startup on e10srollout@mozilla.org version 1.6
1490267391969	addons.xpi	DEBUG	Registering manifest for C:\Program Files\Mozilla Firefox\browser\features\firefox@getpocket.com.xpi
1490267391970	addons.xpi	DEBUG	Calling bootstrap method startup on firefox@getpocket.com version 1.0.5
1490267391970	addons.xpi	DEBUG	Registering manifest for C:\Program Files\Mozilla Firefox\browser\features\webcompat@mozilla.org.xpi
1490267391971	addons.xpi	DEBUG	Calling bootstrap method startup on webcompat@mozilla.org version 1.0
1490267391972	addons.manager	DEBUG	Registering shutdown blocker for XPIProvider
1490267391972	addons.manager	DEBUG	Provider finished startup: XPIProvider
1490267391973	addons.manager	DEBUG	Starting provider: LightweightThemeManager
1490267391973	addons.manager	DEBUG	Registering shutdown blocker for LightweightThemeManager
1490267391973	addons.manager	DEBUG	Provider finished startup: LightweightThemeManager
1490267391974	addons.manager	DEBUG	Starting provider: GMPProvider
1490267391984	addons.manager	DEBUG	Registering shutdown blocker for GMPProvider
1490267391984	addons.manager	DEBUG	Provider finished startup: GMPProvider
1490267391984	addons.manager	DEBUG	Starting provider: PluginProvider
1490267391984	addons.manager	DEBUG	Registering shutdown blocker for PluginProvider
1490267391984	addons.manager	DEBUG	Provider finished startup: PluginProvider
1490267391985	addons.manager	DEBUG	Completed startup sequence
1490267393150	addons.manager	DEBUG	Starting provider: <unnamed-provider>
1490267393150	addons.manager	DEBUG	Registering shutdown blocker for <unnamed-provider>
1490267393151	addons.manager	DEBUG	Provider finished startup: <unnamed-provider>
1490267393152	DeferredSave.extensions.json	DEBUG	Starting write
1490267393628	addons.repository	DEBUG	No addons.json found.
1490267393628	DeferredSave.addons.json	DEBUG	Save changes
1490267393632	DeferredSave.addons.json	DEBUG	Starting timer
1490267393682	addons.manager	DEBUG	Starting provider: PreviousExperimentProvider
1490267393682	addons.manager	DEBUG	Registering shutdown blocker for PreviousExperimentProvider
1490267393682	addons.manager	DEBUG	Provider finished startup: PreviousExperimentProvider
1490267393684	DeferredSave.extensions.json	DEBUG	Write succeeded
1490267393684	addons.xpi-utils	DEBUG	XPI Database saved, setting schema version preference to 19
1490267393684	DeferredSave.addons.json	DEBUG	Starting write
1490267393714	DeferredSave.addons.json	DEBUG	Write succeeded

**Comment by meteboyaci** [comment]
I have passed the issue via Firefox 46.0 version. Firefox 47 and up versions do not work with selenium.
Thnx..

**Comment by bendauphinee** [comment]
I just upgraded to nightly 53.0b5/32bit, and still have the same crash as previous posters. Log dump: https://gist.github.com/bendauphinee/4f9a58d32328195b48101d41a73adb9c

tl;dr: Brower opens fine, sees all elements on first page, navigates to second page, and dies.

**Comment by naveenmails** [comment]
looks like related to #517 . Can some one take a look

**Comment by LinuxSuRen** [comment]
还是没有解决啊

**Comment by QAMiranda** [comment]
having the same problem on windows 10 and firefox 52

1490807822242	geckodriver	INFO	Listening on 127.0.0.1:12819
1490807822923	mozprofile::profile	INFO	Using profile path C:\Users\atallahm\AppData\Local\Temp\rust_mozprofile.AhXAIsiaaW8U
1490807822930	geckodriver::marionette	INFO	Starting browser C:\Program Files (x86)\Mozilla Firefox\firefox.exe with args []
1490807822943	geckodriver::marionette	INFO	Connecting to Marionette on localhost:60663
1490807823834	Marionette	INFO	Listening on port 60663
Mar 29, 2017 11:17:05 AM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
1490807825524	Marionette	INFO	New connections will no longer be accepted
[Child 9636] ###!!! ABORT: Aborting on channel error.: file c:/builds/moz2_slave/m-rel-w32-00000000000000000000/build/src/ipc/glue/MessageChannel.cpp, line 2143
Mar 29, 2017 11:17:07 AM org.openqa.selenium.os.UnixProcess destroy
SEVERE: Unable to kill process with PID 1660

**Comment by andres101** [comment]
Had the same error. Eventually got it working by using these versions:
Firefox Setup 48.0.2 (32bit)
geckodriver-v0.15.0-win32
JavaSetup8u121 (32bit)
selenium-server-standalone-3.3.1

**Comment by danielgonzalezrodero** [comment]
I have a very basic test running and when I just delete the line where the driver quits (driver.quit();), it does not abort the test, but it does not close the firefox window. So I guess this should be related ?

**Comment by cgoldberg** [comment]
@danielgonzalezrodero that's expected behavior.  why would it close the window if you removed the code?

**Comment by danielgonzalezrodero** [comment]
@cgoldberg I explained myself really bad. The ABORTING TEST error is showing up only if I have driver.quit(); line. When I just drop this line it is obvious that it won't close BUT I don't get the ABORT TEST error either. So, that's why I think that driver.quit(); line is causing the error, at least in my machine.

**Comment by andreastt** [comment]
Duplicate of https://github.com/mozilla/geckodriver/issues/285.

**Comment by pinggaimuir** [comment]
我也遇到了同样的问题在selenium3.8.1 ，目前还没有解决办法``
